electron-app/src/logic/sync/domain/use-cases/get-sync-auth.use-case.ts

```diff
@@ -0,0 +1,25 @@
+import { CustomError } from '../../../errors';
+import { ISyncRepository } from '../sync.repository';
+import { getConnector } from '../../../connector/infrastructure/factory';
+
+export interface GetSyncAuthUseCase {
+  execute(id: string): Promise<boolean>;
+}
+
+export class GetSyncAuth implements GetSyncAuthUseCase {
+  constructor(private readonly repository: ISyncRepository) {}
+
+  async execute(id: string) {
+    const data = await this.repository.getSync(id);
```

Comment:
You can create an SyncEntity and get the sourceConector from there to get the auth data. it Is better because we will have only one responsible to manage this, the entity